src/workerd/jsg/modules-new.c++

```diff
@@ -706,76 +714,80 @@ v8::MaybeLocal<v8::Module> resolveCallback(v8::Local<v8::Context> context,
   auto& registry = IsolateModuleRegistry::from(isolate);
   auto& js = Lock::from(isolate);
 
-  try {
-    return js.tryCatch([&]() -> v8::MaybeLocal<v8::Module> {
-      auto spec = kj::str(specifier);
-
-      // The proposed specification for import assertions strongly recommends that
-      // embedders reject import attributes and types they do not understand/implement.
-      // This is because import attributes can alter the interpretation of a module.
-      // Throwing an error for things we do not understand is the safest thing to do
-      // for backwards compatibility.
-      //
-      // For now, we do not support any import attributes, so if there are any at all
-      // we will reject the import.
-      if (!import_assertions.IsEmpty()) {
-        if (import_assertions->Length() > 0) {
-          js.throwException(js.typeError("Import attributes are not supported"));
-        };
-      }
+  return js.tryCatch([&]() -> v8::MaybeLocal<v8::Module> {
+    auto spec = kj::str(specifier);
+
+    // The proposed specification for import assertions strongly recommends that
+    // embedders reject import attributes and types they do not understand/implement.
+    // This is because import attributes can alter the interpretation of a module.
+    // Throwing an error for things we do not understand is the safest thing to do
+    // for backwards compatibility.
+    //
+    // For now, we do not support any import attributes, so if there are any at all
+    // we will reject the import.
+    if (!import_assertions.IsEmpty()) {
+      if (import_assertions->Length() > 0) {
+        js.throwException(js.typeError("Import attributes are not supported"));
+      };
+    }
 
-      ResolveContext::Type type = ResolveContext::Type::BUNDLE;
+    ResolveContext::Type type = ResolveContext::Type::BUNDLE;
 
-      auto& referrerUrl = registry.lookup(js, referrer)
-                              .map([&](IsolateModuleRegistry::Entry& entry) -> const Url& {
-        switch (entry.module.type()) {
-          case Module::Type::BUNDLE: {
-            type = ResolveContext::Type::BUNDLE;
-            break;
-          }
-          case Module::Type::BUILTIN: {
-            type = ResolveContext::Type::BUILTIN;
-            break;
-          }
-          case Module::Type::BUILTIN_ONLY: {
-            type = ResolveContext::Type::BUILTIN_ONLY;
-            break;
-          }
-          case Module::Type::FALLBACK: {
-            type = ResolveContext::Type::BUNDLE;
-            break;
-          }
+    auto& referrerUrl = registry.lookup(js, referrer)
+                            .map([&](IsolateModuleRegistry::Entry& entry) -> const Url& {
+      switch (entry.module.type()) {
+        case Module::Type::BUNDLE: {
```

Comment:
It seems like a hashmap can simplify this code?

Reply:
possibly but seems a bit overkill? This should compile down to something pretty performant without introducing a hashmap

Reply:
i commented for visibility not for performance